common/BasicMultiQueueApplication.hpp

```diff
@@ -121,11 +121,30 @@ class BasicMultiQueueApplication : public virtual MonoDeviceApplication
 		{
 			QueueAllocator queueAllocator(familyProperties);
 
-			// First thing to make sure we have is a compute queue (so nothing else fails allocation) which should be able to do image transfers of any granularity (transfer only queue families can have problems with that)
-			queue_req_t computeQueueRequirement = {.requiredFlags=queue_flags_t::EQF_COMPUTE_BIT,.disallowedFlags=queue_flags_t::EQF_NONE,.queueCount=1,.maxImageTransferGranularity={1,1,1}};
-			m_computeQueue.famIx = queueAllocator.allocateFamily(computeQueueRequirement,{queue_flags_t::EQF_GRAPHICS_BIT,queue_flags_t::EQF_TRANSFER_BIT,queue_flags_t::EQF_SPARSE_BINDING_BIT,queue_flags_t::EQF_PROTECTED_BIT});
-			// since we requested a device that has a compute capable queue family (unless `getQueueRequirements` got overriden) we're sure we'll get at least one family capable of compute
-			assert(m_computeQueue.famIx!=QueueAllocator::InvalidIndex);
+			// This is a sort-of allocator of queue indices for distinct queues
+			core::map<uint8_t, uint8_t> familyQueueCounts;
+
+			if (!isHeadlessCompute())
+			{
+				// TODO: Handle swapchain compatibility.
+				queue_req_t graphicsQueueRequirement = { .requiredFlags = queue_flags_t::EQF_GRAPHICS_BIT,.disallowedFlags = queue_flags_t::EQF_NONE,.queueCount = 1 };
+				// Place queue_flags_t::EQF_NONE in front in case there are no graphics queues without the unwanted flags.
+				m_graphicsQueue.famIx = queueAllocator.allocateFamily(graphicsQueueRequirement, { queue_flags_t::EQF_NONE, queue_flags_t::EQF_TRANSFER_BIT,queue_flags_t::EQF_SPARSE_BINDING_BIT, queue_flags_t::EQF_COMPUTE_BIT, queue_flags_t::EQF_PROTECTED_BIT });
```

Comment:
EQF NONE is just a 0, it won't do anything

Reply:
No it's actually important, being 0 is exactly the point. The way `allocateFamily` works it tries to find a queue without all the flags, then removes one flag and tries again, until it reaches the state with just 1 unwanted flag (the first one). And if that flag is still present in all queues it will fail. That's what's actually happening on my RX 6600 (because there is no queue without the transfer flag). So the last flag `EQF_NONE` is for `allocateFamily` to do one last iteration where it tries to find a graphics queue without any restrictions.

Reply:
ok I see what you mean, the loop will not execute
```cpp
					for (size_t flagCount=unwantedFlags.size(); flagCount; flagCount--)
					{
						queue_req_t req = originalReq;
						for (auto it=unwantedFlags.begin(); it!=(unwantedFlags.begin()+flagCount); it++)
```

I'm rewriting to a `do{}while();`